x/airdrop/keeper/proposal_handler.go

```diff
@@ -49,9 +49,11 @@ func HandleRegisterZoneDropProposal(ctx sdk.Context, k Keeper, p *types.Register
 			return fmt.Errorf("claim record %d, %w", i, err)
 		}
 
-		for j, ca := range cr.ActionsCompleted {
-			if ctx.BlockTime().Before(ca.CompleteTime) {
-				return fmt.Errorf("invalid zonedrop proposal claim record [%d]: completed action [%d] is in future", i, j)
+		if cr.ActionsCompleted != nil {
```

Comment:
even simpler than that surely? we shouldn't accept Proposals with already completed actions? that map should always be nil - thus we should error if it is non-nil?

Reply:
yes indeed, got lost with what I was trying to do...